### DISCUSSION TITLE
How can I install magento2-module-pagebuildersourcecode

### DISCUSSION THREAD

**Discussion by User9511** [markshust/docker-magento#1098] [Q&A]
Hey Guys,

I've  followed these steps to install the magento2-module-pagebuildersourcecode  but i'm getting this error, I've seen some other comments of other  people who are having the same issue, my composer is up to date.

./composer.json has been created
Running composer update markshust/magento2-module-pagebuildersourcecode
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - markshust/magento2-module-pagebuildersourcecode 1.0.0 requires magento/framework 102.0.1.* -> could not be found in any version, there may be a typo in the package name.
    - markshust/magento2-module-pagebuildersourcecode 1.0.1 requires magento/framework 102.0.* -> could not be found in any version, there may be a typo in the package name.
    - markshust/magento2-module-pagebuildersourcecode 1.0.2 requires php ^7.1 -> your php version (8.1.1) does not satisfy that requirement.
    - markshust/magento2-module-pagebuildersourcecode[1.0.3, ..., 1.0.5] require magento/framework >=100 -> could not be found in any version, there may be a typo in the package name.
    - Root composer.json requires markshust/magento2-module-pagebuildersourcecode * -> satisfiable by markshust/magento2-module-pagebuildersourcecode[1.0.0, ..., 1.0.5].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require markshust/magento2-module-pagebuildersourcecode:*" to figure out if any version is installable, or "composer require markshust/magento2-module-pagebuildersourcecode:^2.1" if you know which you need.

Installation failed, deleting ./composer.json.

https://github.com/markshust/magento2-module-pagebuildersourcecode?tab=readme-ov-file#requirements

**Comment by YevhenZvieriev** [comment]
Hi, @User9511 

Can I ask you, if you executed `composer require markshust/magento2-module-pagebuildersourcecode`?

If yes, you should use `bin/composer` instead of `composer`.

So to install `magento2-module-pagebuildersourcecode` you should execute:

1. `bin/composer require markshust/magento2-module-pagebuildersourcecode`

2. `bin/magento module:enable MarkShust_PageBuilderSourceCode`

3. `bin/magento setup:upgrade`
